net/tstun/table/table.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package table provides a Routing Table implementation which allows
+// looking up the peer that should be used to route a given IP address.
+package table
+
+import (
+	"net/netip"
+
+	"tailscale.com/tempfork/device"
+	"tailscale.com/types/key"
+	"tailscale.com/util/mak"
+)
+
+// RoutingTable provides a mapping from IP addresses to peers identified by
+// their public node key. It is used to find the peer that should be used to
+// route a given IP address.
+//
+// It is not safe for concurrent use.
+type RoutingTable struct {
+	peers      map[key.NodePublic]*device.Peer
```

Comment:
Maybe add a comment about why we need peers and that its not the actual routing table?

Reply:
I split it into a RoutingTableBuilder and a RoutingTable, so that its clear that we don't use this field at runtime. Also added docs.